core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -68,6 +68,10 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
         final MediaType contentType = request.contentType();
         if (contentType != null && (contentType.is(MediaType.JSON) ||
                                     contentType.subtype().endsWith("+json"))) {
+            if (expectedResultType == String.class) {
```

Comment:
How would you distinguish between the following two cases when the request body is:

```json
"value"
```

- Case 1: A user expects `"\"value\"", which is an as-is conversion of the request body.
- Case 2: A user expects `"value"`, which is a conversion of the JSON string.

Given the ambiguity, perhaps we should not handle `String.class` specially but only `byte[].class` and `HttpData.class`?